nearcache/config.go

```diff
@@ -0,0 +1,256 @@
+*/
+type EvictionConfig struct {
+	evictionPolicy *EvictionPolicy
+	size           *int32
+	comparator     EvictionPolicyComparator
+	err            error
+}
+
+// Clone returns a copy of the configuration.
+func (c EvictionConfig) Clone() EvictionConfig {
+	return c
+}
+
+// Validate validates the configuration and sets the defaults.
+func (c *EvictionConfig) Validate() error {
+	if c.err != nil {
+		return c.err
+	}
+	if c.evictionPolicy != nil && c.comparator != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.EvictionConfig: only one of EvictionPolicy or Comparator can be configured", nil)
+	}
+	return nil
+}
+
+// SetEvictionPolicy sets the eviction policy of this eviction configuration.
+// The default policy is EvictionPolicyLRU which evicts the least recently used entries.
+func (c *EvictionConfig) SetEvictionPolicy(policy EvictionPolicy) {
+	if policy < 0 || policy >= evictionPolicyCount {
+		c.err = ihzerrors.NewInvalidConfigurationError("nearcache.EvictionConfig.SetEvictionPolicy: invalid policy", nil)
+		return
+	}
+	c.evictionPolicy = &policy
+}
+
+// EvictionPolicy returns the eviction policy of this eviction configuration.
+// See the documentation for SetEvictionPolicy.
+func (c EvictionConfig) EvictionPolicy() EvictionPolicy {
+	if c.evictionPolicy == nil {
+		return defaultEvictionPolicy
+	}
+	return *c.evictionPolicy
+}
+
+// SetSize sets the number of maximum entries before an eviction occurs.
+// Accepts any non-negative number.
+// The default value is 10_000.
+func (c *EvictionConfig) SetSize(size int) {
+	s, err := check.NonNegativeInt32(size)
+	if err != nil {
+		c.err = ihzerrors.NewInvalidConfigurationError("nearcache.EvictionConfig.SetSize: out of range", err)
+		return
+	}
+	c.size = &s
+}
+
+// Size returns the number of maximum entries before an eviction occurs.
+// See the documentation for SetSize.
+func (c EvictionConfig) Size() int {
+	if c.size == nil {
+		return defaultMaxEntryCount
+	}
+	return int(*c.size)
+}
+
+// SetComparator sets the eviction policy comparator.
+func (c *EvictionConfig) SetComparator(cmp EvictionPolicyComparator) {
+	c.comparator = cmp
+}
+
+// Comparator returns the eviction policy comparator.
+func (c EvictionConfig) Comparator() EvictionPolicyComparator {
+	return c.comparator
+}
+
+// PreloaderConfig is the configuration for storing and pre-loading Near Cache keys.
+// Preloader re-populates Near Cache after client restart to provide fast access.
+// Saved preloader data is compatible between only the same versions of Go client.
+// It is disabled by default.
+type PreloaderConfig struct {
+	// Directory is the directory to store preloader cache.
+	Directory string
+	// StoreInitialDelaySeconds is the time in seconds before the preloader starts saving the cache.
+	// Must be positive.
+	// By default it is 600 seconds.
+	StoreInitialDelaySeconds int
+	// StoreIntervalSeconds is the time in seconds for the cache save period.
```

Comment:
```suggestion
	// StoreIntervalSeconds is the interval in seconds for the cache save (operation/task).
```

Reply:
Changed it to: "StoreIntervalSeconds is the interval in seconds for persisting the cache."